### ISSUE TITLE
Peasant spawn speed

### ISSUE THREAD

**Issue by LordUvin** [UnofficialCrusaderPatch/UnofficialCrusaderPatch#1030]
Hi, is it possible to make an option for increasing **peasant spawning speed**? 
As I increased difficulty by using UCP, it is to hard for me to play even on speed 10. Now i have to make sacrifices when playing PvE. 

It would be nice to make it possible **x2 or 3x or even x4 speed increase** for peasant spawn.

I have another  idea- it would be awesome to have **infinite peasants at campfire**. As you build or recruit, the number of peasants don't go down. But there would be one problem. What if a building near the camp fire starts to burn. Then all around the camp fire would spawn burning peasants. Hmm it is just an idea. 

What do you think about it?

![just_adding_a_picture_nothing _special](https://github.com/UnofficialCrusaderPatch/UnofficialCrusaderPatch/assets/87543244/f0e98cfa-2e5d-4a4e-9159-6e7edd1df031)

**Comment by LordUvin** [comment]
> Peasant spawn speed depends on your popularity as a Lord. 100 popularity means they spawn at the quickest rate. Would you want to increase spawn speed for any level of popularity? Like x2 overall?
> 
> Infinite peasants at the campfire is not possible, you would need to get rid of the "crowding" popularity penalty as well. You would need infinite food as well to support the infinite amount of peasants. The amount of peasants is tied to the amount of hovels you build. When playing at higher speeds and building more hovels, the amount of peasants will replenish faster than you can recruit units (giving the feel of infinite peasants).


Yes basically I mean overall speed increase by x2 or x3 or x4 would be nice.
By infinite I mean spawning speed. I agree, peasants could only be tied to the amount of hovels built.

When you said
"
When playing at higher speeds and building more hovels, the amount of peasants will replenish faster than you can recruit units (giving the feel of infinite peasants).
"
I understand it very well. When I play mission 19 "Crossroads" at Crusader Extreme Trail. I need speed 10 till I start to attack. After that 35 speed. (I don't want to flex, but I have played this mission about 100 hours. Very nice, I like it.

P.S.
I always try to play with 100 popularity!

**Comment by gynt** [comment]
Do you maybe want to share your .sav file of that mission? Would love to try that one too, but I don't want to bother with playing the missions 1 to 18 :)

**Comment by LordUvin** [comment]
> Do you maybe want to share your .sav file of that mission? Would love to try that one too, but I don't want to bother with playing the missions 1 to 18 :)

Yes, okay.
https://drive.google.com/file/d/1MQ5k307BD7rRjGWHTIc0hAinlql2atsm/view?usp=sharing

I could even help with tips. If you want some tips, just ask.

**Comment by SahandAslani** [comment]
change the second operand of the compare instruction at 0x045b587 it's 0x0000FA0(4000) change it to  0x00007D0(2000) or even 0x0000001 and you won't need to wait

**Comment by LordUvin** [comment]
Thank you for your reply

**Comment by SahandAslani** [comment]
for number of peasants change the operand of 0x045b5a2 it's 24

**Comment by gynt** [comment]
For a future dev:
```lua
--[[
For Skirmish mode, sets the positive increments of the peasant spawn clock for positive popularity to 4000 (max value, resulting in instant respawning of peasants).
--]]
for i=10, 20 do
  core.writeInteger(core.readInteger(core.AOBScan("8B ? ? ? ? ? ? EB 2D") + 3) + (4*i), 4000)
end
```
